client/ayon_flame/plugins/create/create_shot_clip.py

```diff
@@ -1,51 +1,555 @@
+            except KeyError:
+                marker_data[_CONTENT_ID] = {}
+                instances_data = marker_data[_CONTENT_ID]
+
+            instances_data[self.identifier] = created_inst.data_to_store()
+            pipeline.imprint(
+                segment_item,
+                data=  {
+                    _CONTENT_ID: instances_data,
+                    "clip_index": marker_data["clip_index"],
+                }
+            )
+
+    def remove_instances(self, instances):
+        """Remove instance marker from track item.
+
+        Args:
+            instance(List[CreatedInstance]): Instance objects which should be
+                removed.
+        """
+        for instance in instances:
+            segment_item = instance.transient_data["segment_item"]
+            marker_data = ayfapi.get_segment_data_marker(segment_item)
+
+            instances_data = marker_data.get(_CONTENT_ID, {})
+            instances_data.pop(self.identifier, None)
+            self._remove_instance_from_context(instance)
+
+            pipeline.imprint(
+                segment_item,
+                data=  {
+                    _CONTENT_ID: instances_data,
+                    "clip_index": marker_data["clip_index"],
+                }
+            )
+
+
+class FlameShotInstanceCreator(_FlameInstanceCreator):
+    """Shot product type creator class"""
+    identifier = "io.ayon.creators.flame.shot"
+    product_type = "shot"
+    label = "Editorial Shot"
+
+    def get_instance_attr_defs(self):
+        instance_attributes = CLIP_ATTR_DEFS
+        return instance_attributes
+
+
+class _FlameInstanceClipCreatorBase(_FlameInstanceCreator):
+    """ Base clip product creator.
+    """
 
+    def register_callbacks(self):
+        self.create_context.add_value_changed_callback(self._on_value_change)
 
-class CreateShotClip(ayfapi.Creator):
+    def _on_value_change(self, event):
+        for item in event["changes"]:
+            instance = item["instance"]
+            if (
+                instance is None
+                or instance.creator_identifier != self.identifier
+            ):
+                continue
+
+            changes = item["changes"].get("creator_attributes", {})
+            if "review" not in changes:
+                continue
+
+            attr_defs = instance.creator_attributes.attr_defs
+            review_value = changes["review"]
+            reviewable_source = next(
+                attr_def
+                for attr_def in attr_defs
+                if attr_def.key == "reviewTrack"
+            )
+            reviewable_source.enabled = review_value
+
+            instance.set_create_attr_defs(attr_defs)
+
+    def get_attr_defs_for_instance(self, instance):
+
+        current_sequence = lib.get_current_sequence(lib.CTX.selection)
+        if current_sequence is not None:
+            gui_tracks = get_video_track_names(current_sequence)
+        else:
+            gui_tracks = []
```

Comment:
Another distinque difference is that in Hiero we have moved from **List[str]** towards **List[dict]** so we are able to improve labeling for better user communication. Look here https://github.com/ynput/ayon-hiero/blob/develop/client/ayon_hiero/plugins/create/create_shot_clip.py#L211-L216

Reply:
Thanks, as suggested I'll look into this as a separate PR.